tests/testthat/test-assemble.R

```diff
@@ -0,0 +1,26 @@
+no_dupes = cars[1, ]
+dupes = rbind(no_dupes, no_dupes)
+
+test_that("detect duplication", {
+  expect_true(contains_duplicates(dupes))
+  expect_false(contains_duplicates(no_dupes))
+})
+
+test_that("keep duplication", {
+  expect_warning(res <- duplicate_keep(dupes))
+  expect_identical(res, dupes)
+  expect_warning(res <- duplicate_keep(no_dupes), NA) # expect no warning
```

Comment:
Could this be `expect_silent()` instead? No need to change but just a suggestion

Reply:
As far as i understand `expect_silent()` expects no errors, warnings or messages so I deliberately opted for the expect no warning as that is really what I am expecting